homeassistant/helpers/service.py

```diff
@@ -109,3 +110,41 @@ def extract_entity_ids(hass, service_call, expand_group=True):
             return [service_ent_id]
 
         return service_ent_id
+
+
+@callback
+def async_get_state_services(hass, domain, state):
+    """Return a dict of services for domain that can be used to set state.
+
+    This function must be run in the event loop.
+    """
+    domain_services = hass.services.async_full_services().get(domain)
```

Comment:
Why not allow passing in the domain directly to `async_services` ? The sync version won't be able to support that, but that's fine.